controllers/ajax.py

```diff
@@ -27,9 +27,10 @@
                'clickablearea':'clickablearea_answers',
                'parsonsprob': 'parsons_answers' }
 
+
 def compareAndUpdateCookieData(sid):
     if request.cookies.has_key('ipuser') and request.cookies['ipuser'].value != sid:
-        db.useinfo(db.useinfo.sid == request.cookies['ipuser'].value).update(sid=sid)
+        db.useinfo.update_or_insert(db.useinfo.sid == request.cookies['ipuser'].value, sid=sid)
```

Comment:
Do we really want update or insert here??  I think update should be sufficient.

Reply:
In the case that this is the first time this `sid` is encountered, then this must be an update. If the `sid` is already in the database, an update works fine. So, I have seen failures when the first case is encountered.